core/src/impl/Kokkos_FunctorAnalysis.hpp

```diff
@@ -673,6 +678,11 @@ struct FunctorAnalysis {
                                          detected_volatile_join_tag<F>::value)>>
       : public has_volatile_join_tag_function<F> {
     enum : bool { value = true };
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_3
```

Comment:
I failed to record whether we wanted this to be 3 or 4

Reply:
And, whether we wanted to get this into 3.7

Reply:
You got it right.  I think this missed the train as #5408 is open now.  @crtrott what do you think?

Reply:
FYI - https://github.com/kokkos/kokkos-kernels/runs/8123690491?check_suite_focus=true